shim.c

```diff
@@ -1377,6 +1377,84 @@ uninstall_shim_protocols(void)
 #endif
 }
 
+void
+check_section(char *section_name, int len, void **pointer,
+		EFI_IMAGE_SECTION_HEADER *Section, void *data, int datasize)
+{
+	if (CompareMem(Section->Name, section_name, len) == 0) {
+		*pointer = ImageAddress(data, datasize, Section->PointerToRawData);
+		if (!*pointer) {
+			return ;
+		}
+		dprint(L"found %.*s\n", len, section_name);
+	}
```

Comment:
I can add those checks, but does it really make sense to validate that a signed binary isn't broken?
I suppose someone could sign a negative test and ship that by mistake.

Reply:
Since I couldn't immediately figure out a way to do this at build time in the mule binary, I added some checks here. I didn't limit the sbat payload size, although we may want to come up with a worst case number for that as well.

I'm still thinking about a way to assert this at build time in the mule as well.